File: 69.sqrt-x.py
#
# @lc app=leetcode id=69 lang=python3
#
# [69] Sqrt(x)
#

# @lc code=start
class Solution:
    def mySqrt(self, a: int) -> int:
        x = 10**-3
        for _ in range(29):
            x -= x/2 - a/(2*x)
        return int(x)
# @lc code=end

File: 69.sqrt-x.rs
/*
 * @lc app=leetcode id=69 lang=rust
 *
 * [69] Sqrt(x)
 */

// @lc code=start
impl Solution {
    pub fn my_sqrt(a: i32) -> i32 {
        let mut x = a as f64;
        for _ in 0..20 {
            x -= x/2.0-a as f64/(2.0*x);
        }
        
        return x as i32;
    }
}
// @lc code=end

